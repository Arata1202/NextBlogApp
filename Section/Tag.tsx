export const tags = [
  { name: 'Git', link: '/tag/git' },
  { name: 'SEO対策', link: '/tag/seo' },
];
