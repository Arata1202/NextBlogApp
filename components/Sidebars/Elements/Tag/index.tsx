import { HashtagIcon } from '@heroicons/react/24/solid';
import styles from './index.module.css';
import { tags } from '@/section/tag';

export default function Tag() {
  return (
    <div className="bg-white pt-8 px-4 border border-gray-300 py-5 mt-5">
      <h1 className={`${styles.profile} text-2xl text-center font-semibold flex justify-center`}>
        <HashtagIcon className="h-8 w-8 mr-2" aria-hidden="true" />
        タグ
      </h1>
      <div className="mt-5 flex flex-wrap gap-2">
        {tags.map((tag, index) => (
          <a
            key={index}
            href={tag.link}
            className="inline-block border border-gray-300 rounded-full px-3 py-1 text-sm font-semibold text-gray-900 mr-2 mb-2 hover:text-blue-500"
          >
            {tag.name}
          </a>
        ))}
      </div>
    </div>
  );
}
