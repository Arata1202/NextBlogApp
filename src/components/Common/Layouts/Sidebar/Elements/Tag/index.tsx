'use client';

import { useTheme } from 'next-themes';
import { HashtagIcon } from '@heroicons/react/24/solid';
import { tags } from '@/section/tag';

export default function Tag() {
  const { theme } = useTheme();

  return (
    <>
      <div
        className={`pt-8 px-4 border py-5 mt-5 ${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}
      >
        <div className={`text-2xl text-center font-semibold flex justify-center`}>
          <HashtagIcon className="h-8 w-8 mr-2" />
          タグ
        </div>
        <div className="mt-5 flex flex-wrap gap-2">
          {tags.map((tag) => (
            <a
              key={tag.id}
              href={`/tag/${tag.id}`}
              className={`inline-block border rounded-full px-3 py-1 text-sm font-semibold mr-2 mb-2 hover:text-blue-500 hover:border-blue-500 ${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}
            >
              {tag.name}
            </a>
          ))}
        </div>
      </div>
    </>
  );
}
