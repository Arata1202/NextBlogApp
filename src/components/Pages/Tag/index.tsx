import { Article, Tag } from '@/libs/Microcms';
import PageHeading from '@/components/Common/PageHeading';
import Pagination from '@/components/Common/Pagination';
import ArticleList from '@/components/Common/ArticleList';
import AdUnit from '@/components/ThirdParties/GoogleAdSense/Elements/AdUnit';

type Props = {
  articles: Article[];
  tag: Tag;
  current?: number;
  totalCount: number;
  recentArticles?: Article[];
};

export default function TagPage({ articles, tag, totalCount, recentArticles, current }: Props) {
  return (
    <>
      <PageHeading tag={tag} />
      <ArticleList articles={articles} recentArticles={recentArticles} />
      <Pagination totalCount={totalCount} current={current} basePath={`/tag/${tag.id}`} />
      <AdUnit slot="5969933704" style={{ marginTop: '1.25rem' }} />
    </>
  );
}
