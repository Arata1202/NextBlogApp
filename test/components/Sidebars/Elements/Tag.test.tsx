import { render } from '@testing-library/react';
import Tag from '@/components/Sidebars/Elements/Tag';

describe('Tag', () => {
  test('スナップショット（Tag）', () => {
    const { asFragment } = render(<Tag />);
    expect(asFragment()).toMatchSnapshot();
  });
});
