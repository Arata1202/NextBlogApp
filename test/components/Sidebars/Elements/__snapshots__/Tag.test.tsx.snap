// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag スナップショット（Tag） 1`] = `
<DocumentFragment>
  <div
    class="bg-white pt-8 px-4 border border-gray-300 py-5 mt-5"
  >
    <h1
      class="profile text-2xl text-center font-semibold flex justify-center"
    >
      <svg
        aria-hidden="true"
        class="h-8 w-8 mr-2"
        data-slot="icon"
        fill="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M11.097 1.515a.75.75 0 0 1 .589.882L10.666 7.5h4.47l1.079-5.397a.75.75 0 1 1 1.47.294L16.665 7.5h3.585a.75.75 0 0 1 0 1.5h-3.885l-1.2 6h3.585a.75.75 0 0 1 0 1.5h-3.885l-1.08 5.397a.75.75 0 1 1-1.47-.294l1.02-5.103h-4.47l-1.08 5.397a.75.75 0 1 1-1.47-.294l1.02-5.103H3.75a.75.75 0 0 1 0-1.5h3.885l1.2-6H5.25a.75.75 0 0 1 0-1.5h3.885l1.08-5.397a.75.75 0 0 1 .882-.588ZM10.365 9l-1.2 6h4.47l1.2-6h-4.47Z"
          fill-rule="evenodd"
        />
      </svg>
      タグ
    </h1>
    <div
      class="mt-5 flex flex-wrap gap-2"
    >
      <a
        class="inline-block border border-gray-300 rounded-full px-3 py-1 text-sm font-semibold text-gray-900 mr-2 mb-2 hover:text-blue-500"
        href="/tag/seo"
      >
        SEO対策
      </a>
      <a
        class="inline-block border border-gray-300 rounded-full px-3 py-1 text-sm font-semibold text-gray-900 mr-2 mb-2 hover:text-blue-500"
        href="/tag/microcms"
      >
        MicroCMS
      </a>
      <a
        class="inline-block border border-gray-300 rounded-full px-3 py-1 text-sm font-semibold text-gray-900 mr-2 mb-2 hover:text-blue-500"
        href="/tag/adsense"
      >
        Google AdSense
      </a>
      <a
        class="inline-block border border-gray-300 rounded-full px-3 py-1 text-sm font-semibold text-gray-900 mr-2 mb-2 hover:text-blue-500"
        href="/tag/git"
      >
        Git
      </a>
      <a
        class="inline-block border border-gray-300 rounded-full px-3 py-1 text-sm font-semibold text-gray-900 mr-2 mb-2 hover:text-blue-500"
        href="/tag/javascript"
      >
        JavaScript
      </a>
      <a
        class="inline-block border border-gray-300 rounded-full px-3 py-1 text-sm font-semibold text-gray-900 mr-2 mb-2 hover:text-blue-500"
        href="/tag/finance"
      >
        金融
      </a>
      <a
        class="inline-block border border-gray-300 rounded-full px-3 py-1 text-sm font-semibold text-gray-900 mr-2 mb-2 hover:text-blue-500"
        href="/tag/aws"
      >
        AWS
      </a>
      <a
        class="inline-block border border-gray-300 rounded-full px-3 py-1 text-sm font-semibold text-gray-900 mr-2 mb-2 hover:text-blue-500"
        href="/tag/cook"
      >
        料理
      </a>
      <a
        class="inline-block border border-gray-300 rounded-full px-3 py-1 text-sm font-semibold text-gray-900 mr-2 mb-2 hover:text-blue-500"
        href="/tag/appledeveloper"
      >
        Apple Developer
      </a>
      <a
        class="inline-block border border-gray-300 rounded-full px-3 py-1 text-sm font-semibold text-gray-900 mr-2 mb-2 hover:text-blue-500"
        href="/tag/gadget"
      >
        ガジェット
      </a>
      <a
        class="inline-block border border-gray-300 rounded-full px-3 py-1 text-sm font-semibold text-gray-900 mr-2 mb-2 hover:text-blue-500"
        href="/tag/flutter"
      >
        Flutter
      </a>
    </div>
  </div>
</DocumentFragment>
`;
